Clang warnings -Winconsistent-missing-override in the Polyhedron Demo

## Issue Details

Since the PR #1972, there are warnings in the testsuite in the clang columns. Since clang 3.5, `-Wall` implies the warning flag [`-Winconsistent-missing-override`][1].

  [1]: https://clang.llvm.org/docs/DiagnosticsReference.html#winconsistent-missing-override

See for example the warnings in https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.10-Ic-171/Installation/TestReport_lrineau_ArchLinux-clang-CXX14.gz

To reproduce those warnings, you need to compile the polyhedron demo, and only the following items:
  - 1 warning: `Scene_polyhedron_item.h`
  - 11 warnings: `Scene_polygon_soup_item.h`
  - 11 warnings: `Scene_surface_mesh_item.h`
  - 13 warnings: `Scene_points_with_normal_item.h`
  - 21 warnings: `Scene_c3t3_item.h`

with either clang>=3.5 and the warning flag `-Wall`, or gcc>=5.1 and the warning flag `-Wsuggest-override`.

Here is a summary of the current list of warnings (each line is a missing [`Q_DECL_OVERRIDE`][2]:
```
Scene_c3t3_item.h:102:20: warning: 'clone' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:112:11: warning: 'toolTip' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:115:8: warning: 'supportsRenderingMode' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:119:8: warning: 'draw' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:120:8: warning: 'drawEdges' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:121:8: warning: 'drawPoints' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:123:12: warning: 'contextMenu' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:124:10: warning: 'copyProperties' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:126:10: warning: 'keyPressEvent' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:140:19: warning: 'graphicalToolTip' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:43:8: warning: 'has_stats' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:44:11: warning: 'computeStats' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:45:40: warning: 'header' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:48:8: warning: 'setColor' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:60:8: warning: 'invalidateOpenGLBuffers' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:71:8: warning: 'manipulatable' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:80:21: warning: 'manipulatedFrame' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:88:8: warning: 'isFinite' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:89:8: warning: 'isEmpty' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:97:8: warning: 'compute_bbox' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_c3t3_item.h:98:20: warning: 'bbox' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:29:34: warning: 'clone' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:35:10: warning: 'contextMenu' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:46:19: warning: 'toolTip' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:48:16: warning: 'invalidateOpenGLBuffers' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:51:16: warning: 'supportsRenderingMode' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:53:16: warning: 'drawEdges' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:54:16: warning: 'drawPoints' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:56:16: warning: 'drawSplats' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:63:16: warning: 'isFinite' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:64:16: warning: 'isEmpty' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:65:16: warning: 'compute_bbox' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:67:16: warning: 'setRenderingMode' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_points_with_normal_item.h:73:8: warning: 'copyProperties' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:113:30: warning: 'clone' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:125:13: warning: 'toolTip' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:128:18: warning: 'supportsRenderingMode' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:130:18: warning: 'draw' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:131:18: warning: 'draw' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:132:18: warning: 'drawPoints' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:133:18: warning: 'drawEdges' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:134:10: warning: 'invalidateOpenGLBuffers' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:135:10: warning: 'isFinite' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:136:10: warning: 'isEmpty' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polygon_soup_item.h:137:10: warning: 'compute_bbox' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_polyhedron_item.h:84:10: warning: 'draw' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:41:28: warning: 'clone' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:42:8: warning: 'draw' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:43:8: warning: 'drawEdges' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:44:8: warning: 'drawPoints' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:46:8: warning: 'supportsRenderingMode' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:47:8: warning: 'isFinite' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:48:8: warning: 'isEmpty' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:49:8: warning: 'bbox' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:50:11: warning: 'toolTip' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:54:8: warning: 'compute_bbox' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
Scene_surface_mesh_item.h:58:16: warning: 'selection_changed' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
```

  [2]: http://doc.qt.io/qt-5/qtglobal.html#Q_DECL_OVERRIDE

--- a/.travis/build_package.sh
+++ b/.travis/build_package.sh
@@ -1,7 +1,53 @@
 #!/bin/bash
 set -e
+
+CXX_FLAGS="-DCGAL_NDEBUG"
+
+function build_examples {
+  mkdir -p build-travis
+  cd build-travis
+  cmake -DCGAL_DIR="$ROOT/build" -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS}" ..
+  make -j2
+}
+
+function build_tests {
+  build_examples
+}
+
+function build_demo {
+  mkdir -p build-travis
+  cd build-travis
+  if [ $NEED_3D = 1 ]; then
+    #install libqglviewer
+    git clone --depth=1 https://github.com/GillesDebunne/libQGLViewer.git ./qglviewer
+    pushd ./qglviewer/QGLViewer
+    #use qt5 instead of qt4
+    export QT_SELECT=5
+    qmake NO_QT_VERSION_SUFFIX=yes
+    make -j2
+    if [ ! -f libQGLViewer.so ]; then
+        echo "libQGLViewer.so not made"
+        exit 1
+    else
+      echo "QGLViewer built successfully"
+    fi
+    #end install qglviewer
+    popd
+  fi
+  EXTRA_CXX_FLAGS=
+  if [ "$CC" = "clang" ]; then
+    EXTRA_CXX_FLAGS="-Werror=inconsistent-missing-override"
+  fi
+  if [ $NEED_3D = 1 ]; then
+    QGLVIEWERROOT=$PWD/qglviewer
+    export QGLVIEWERROOT
+  fi
+  cmake -DCGAL_DIR="$ROOT/build" -DCGAL_DONT_OVERRIDE_CMAKE_FLAGS:BOOL=ON -DCMAKE_CXX_FLAGS_RELEASE="${CXX_FLAGS} ${EXTRA_CXX_FLAGS}" ..
+  make -j2
+}
+
 IFS=$' '
-ROOT="$PWD/../"
+ROOT="$PWD/.."
 NEED_3D=0
 for ARG in $(echo "$@")
 do
@@ -56,21 +102,15 @@ do
 	if [ -d "$ROOT/$EXAMPLES" ]
 	then
 	  cd $ROOT/$EXAMPLES
-	  mkdir -p build
-	  cd build
-	  cmake -DCGAL_DIR="$ROOT/build" -DCMAKE_CXX_FLAGS_RELEASE="-DCGAL_NDEBUG" ..
-	  make -j2
+	  build_examples
   elif [ "$ARG" != Polyhedron_demo ]; then
     echo "No example found for $ARG"
 	fi
 
 	if [ -d "$ROOT/$TEST" ]
 	then
-	  cd $ROOT/$TEST
-	  mkdir -p build
-	  cd build
-	  cmake -DCGAL_DIR="$ROOT/build" -DCMAKE_CXX_FLAGS_RELEASE="-DCGAL_NDEBUG" ..
-  	make -j2
+    cd $ROOT/$TEST
+    build_tests
   elif [ "$ARG" != Polyhedron_demo ]; then
     echo "No test found for $ARG"
 	fi
@@ -84,55 +124,17 @@ do
     fi
 	  if [ "$ARG" != Polyhedron ] && [ -d "$ROOT/$DEMO" ]
   	then
-      if [ $NEED_3D = 1 ]; then
-    	  cd $ROOT/$DEMO
-        #install libqglviewer
-        git clone --depth=1 https://github.com/GillesDebunne/libQGLViewer.git ./qglviewer
-        cd ./qglviewer/QGLViewer
-        #use qt5 instead of qt4
-        export QT_SELECT=5
-        qmake NO_QT_VERSION_SUFFIX=yes
-        make -j2
-        if [ ! -f libQGLViewer.so ]; then
-          echo "libQGLViewer.so not made"
-          exit 1
-        else
-          echo "QGLViewer built successfully"
-        fi
-        #end install qglviewer
-      fi
-	    cd $ROOT/$DEMO
-	    mkdir -p build
-	    cd build
-      if [ $NEED_3D = 1 ]; then
-	      cmake -DCGAL_DIR="$ROOT/build" -DQGLVIEWER_INCLUDE_DIR="$ROOT/$DEMO/qglviewer" -DQGLVIEWER_LIBRARIES="$ROOT/$DEMO/qglviewer/QGLViewer/libQGLViewer.so" -DCMAKE_CXX_FLAGS_RELEASE="-DCGAL_NDEBUG" ..
-      else
-        cmake -DCGAL_DIR="$ROOT/build" -DCMAKE_CXX_FLAGS_RELEASE="-DCGAL_NDEBUG" ..
-      fi
-	    make -j2
+      cd $ROOT/$DEMO
+      build_demo
     elif [ "$ARG" != Polyhedron_demo ]; then
       echo "No demo found for $ARG"
 	  fi
   done
   if [ "$ARG" = Polyhedron_demo ]; then
-    cd "$ROOT/Polyhedron/demo/Polyhedron"
-    #install libqglviewer
-    git clone --depth=1 https://github.com/GillesDebunne/libQGLViewer.git ./qglviewer
-    cd ./qglviewer/QGLViewer
-    #use qt5 instead of qt4
-    export QT_SELECT=5
-    qmake NO_QT_VERSION_SUFFIX=yes
-    make -j2
-    if [ ! -f libQGLViewer.so ]; then
-      echo "libQGLViewer.so not made"
-      exit 1
-    fi
-    #end install qglviewer
-    cd "$ROOT/Polyhedron/demo/Polyhedron"
-    mkdir -p ./build
-    cd ./build
-    cmake -DCGAL_DIR="$ROOT/build" -DQGLVIEWER_INCLUDE_DIR="$ROOT/Polyhedron/demo/Polyhedron/qglviewer" -DQGLVIEWER_LIBRARIES="$ROOT/Polyhedron/demo/Polyhedron/qglviewer/QGLViewer/libQGLViewer.so" -DCMAKE_CXX_FLAGS_RELEASE="-DCGAL_NDEBUG" ..
-    make -j2
+    DEMO=Polyhedron/demo/Polyhedron
+    NEED_3D=1
+    cd "$ROOT/$DEMO"
+    build_demo
   fi  
 done
 

--- a/Polyhedron/demo/Polyhedron/Scene_c3t3_item.h
+++ b/Polyhedron/demo/Polyhedron/Scene_c3t3_item.h
@@ -40,12 +40,12 @@ public:
   Scene_c3t3_item(const C3t3& c3t3);
   ~Scene_c3t3_item();
 
-  bool has_stats()const {return true;}
-  QString computeStats(int type);
-  CGAL::Three::Scene_item::Header_data header() const;
+  bool has_stats()const  Q_DECL_OVERRIDE {return true;}
+  QString computeStats(int type)  Q_DECL_OVERRIDE;
+  CGAL::Three::Scene_item::Header_data header() const Q_DECL_OVERRIDE;
 
 
-  void setColor(QColor c);
+  void setColor(QColor c) Q_DECL_OVERRIDE;
   bool save_binary(std::ostream& os) const
   {
     return CGAL::Mesh_3::save_binary_file(os, c3t3());
@@ -57,7 +57,7 @@ public:
       return !!(os << c3t3());
   }
 
-  void invalidateOpenGLBuffers();
+  void invalidateOpenGLBuffers() Q_DECL_OVERRIDE;
 
   void c3t3_changed();
 
@@ -68,7 +68,7 @@ public:
   const C3t3& c3t3() const;
   C3t3& c3t3();
 
-  bool manipulatable() const {
+  bool manipulatable() const  Q_DECL_OVERRIDE{
     return true;
   }
 
@@ -77,7 +77,7 @@ public:
   bool has_cnc() const;
   bool has_tets() const;
   bool is_valid() const;//true if the c3t3 is correct, false if it was made from a .mesh, for example
-  ManipulatedFrame* manipulatedFrame();
+  ManipulatedFrame* manipulatedFrame() Q_DECL_OVERRIDE;
 
   void setPosition(float x, float y, float z) ;
 
@@ -85,8 +85,8 @@ public:
 
   Kernel::Plane_3 plane(qglviewer::Vec offset = qglviewer::Vec(0,0,0)) const;
 
-  bool isFinite() const { return true; }
-  bool isEmpty() const {
+  bool isFinite() const Q_DECL_OVERRIDE { return true; }
+  bool isEmpty() const Q_DECL_OVERRIDE {
     return c3t3().triangulation().number_of_vertices() == 0
       || (    c3t3().number_of_vertices_in_complex() == 0
            && c3t3().number_of_facets_in_complex()   == 0
@@ -94,12 +94,12 @@ public:
   }
 
 
-  void compute_bbox() const;
-  Scene_item::Bbox bbox() const
+  void compute_bbox() const Q_DECL_OVERRIDE;
+  Scene_item::Bbox bbox() const Q_DECL_OVERRIDE
   {
       return Scene_item::bbox();
   }
-  Scene_c3t3_item* clone() const {
+  Scene_c3t3_item* clone() const  Q_DECL_OVERRIDE{
     return 0;
   }
 
@@ -109,21 +109,21 @@ public:
   const Scene_item* data_item() const;
   void set_data_item(const Scene_item* data_item);
 
-  QString toolTip() const;
+  QString toolTip() const Q_DECL_OVERRIDE;
 
   // Indicate if rendering mode is supported
-  bool supportsRenderingMode(RenderingMode m) const {
+  bool supportsRenderingMode(RenderingMode m) const  Q_DECL_OVERRIDE{
     return (m != Gouraud && m != PointsPlusNormals && m != Splatting && m != Points && m != ShadedPoints);
   }
 
-  void draw(CGAL::Three::Viewer_interface* viewer) const;
-  void drawEdges(CGAL::Three::Viewer_interface* viewer) const;
-  void drawPoints(CGAL::Three::Viewer_interface * viewer) const;
+  void draw(CGAL::Three::Viewer_interface* viewer) const Q_DECL_OVERRIDE;
+  void drawEdges(CGAL::Three::Viewer_interface* viewer) const Q_DECL_OVERRIDE;
+  void drawPoints(CGAL::Three::Viewer_interface * viewer) const Q_DECL_OVERRIDE;
   public:
-    QMenu* contextMenu();
-    void copyProperties(Scene_item *);
+    QMenu* contextMenu() Q_DECL_OVERRIDE;
+    void copyProperties(Scene_item *) Q_DECL_OVERRIDE;
     float getShrinkFactor() const;
-    bool keyPressEvent(QKeyEvent *);
+    bool keyPressEvent(QKeyEvent *) Q_DECL_OVERRIDE;
   public Q_SLOTS:
   void export_facets_in_complex();
 
@@ -137,7 +137,7 @@ public:
   void show_grid(bool b);
   void show_cnc(bool b);
 
-  virtual QPixmap graphicalToolTip() const;
+  virtual QPixmap graphicalToolTip() const Q_DECL_OVERRIDE;
 
   void update_histogram();
 

--- a/Polyhedron/demo/Polyhedron/Scene_points_with_normal_item.h
+++ b/Polyhedron/demo/Polyhedron/Scene_points_with_normal_item.h
@@ -26,13 +26,13 @@ public:
   Scene_points_with_normal_item(const Scene_points_with_normal_item& toCopy);
   Scene_points_with_normal_item(const Polyhedron& p);
   ~Scene_points_with_normal_item();
-  Scene_points_with_normal_item* clone() const;
+  Scene_points_with_normal_item* clone() const Q_DECL_OVERRIDE;
 
   // Is selection empty?
   virtual bool isSelectionEmpty() const;
 
   // Function to override the context menu
-  QMenu* contextMenu();
+  QMenu* contextMenu() Q_DECL_OVERRIDE;
 
   // IO
   bool read_ply_point_set(std::istream& in);
@@ -43,34 +43,34 @@ public:
   bool write_xyz_point_set(std::ostream& out) const;
 
   // Function for displaying meta-data of the item
-  virtual QString toolTip() const;
+  virtual QString toolTip() const Q_DECL_OVERRIDE;
 
-  virtual void invalidateOpenGLBuffers();
+  virtual void invalidateOpenGLBuffers() Q_DECL_OVERRIDE;
 
   // Indicate if rendering mode is supported
-  virtual bool supportsRenderingMode(RenderingMode m) const;
+  virtual bool supportsRenderingMode(RenderingMode m) const Q_DECL_OVERRIDE;
 
-  virtual void drawEdges(CGAL::Three::Viewer_interface* viewer) const;
-  virtual void drawPoints(CGAL::Three::Viewer_interface*) const;
+  virtual void drawEdges(CGAL::Three::Viewer_interface* viewer) const Q_DECL_OVERRIDE;
+  virtual void drawPoints(CGAL::Three::Viewer_interface*) const Q_DECL_OVERRIDE;
 
-  virtual void drawSplats(CGAL::Three::Viewer_interface*) const;
+  virtual void drawSplats(CGAL::Three::Viewer_interface*) const Q_DECL_OVERRIDE;
   
   // Gets wrapped point set
   Point_set*       point_set();
   const Point_set* point_set() const;
 
   // Gets dimensions
-  virtual bool isFinite() const { return true; }
-  virtual bool isEmpty() const;
-  virtual void compute_bbox() const;
+  virtual bool isFinite() const Q_DECL_OVERRIDE { return true; }
+  virtual bool isEmpty() const Q_DECL_OVERRIDE;
+  virtual void compute_bbox() const Q_DECL_OVERRIDE;
 
-  virtual void setRenderingMode(RenderingMode m);
+  virtual void setRenderingMode(RenderingMode m) Q_DECL_OVERRIDE;
 
   // computes the local point spacing (aka radius) of each point
   void computes_local_spacing(int k);
 
   bool has_normals() const;
-  void copyProperties(Scene_item *);
+  void copyProperties(Scene_item *) Q_DECL_OVERRIDE;
   int getNormalSliderValue();
   int getPointSliderValue();
 

--- a/Polyhedron/demo/Polyhedron/Scene_polygon_soup_item.h
+++ b/Polyhedron/demo/Polyhedron/Scene_polygon_soup_item.h
@@ -110,7 +110,7 @@ public:
     Scene_polygon_soup_item();
     ~Scene_polygon_soup_item();
 
-    Scene_polygon_soup_item* clone() const;
+    Scene_polygon_soup_item* clone() const Q_DECL_OVERRIDE;
 
     template <class Point, class Polygon>
     void load(const std::vector<Point>& points, const std::vector<Polygon>& polygons);
@@ -122,19 +122,19 @@ public:
     bool save(std::ostream& out) const;
     std::vector<CGAL::Color> getVColors() const;
     std::vector<CGAL::Color> getFColors() const;
-    QString toolTip() const;
+    QString toolTip() const Q_DECL_OVERRIDE;
 
     // Indicate if rendering mode is supported
-    virtual bool supportsRenderingMode(RenderingMode m) const { return ( m!=PointsPlusNormals && m!=Splatting && m!=ShadedPoints); }
+    virtual bool supportsRenderingMode(RenderingMode m) const Q_DECL_OVERRIDE{ return ( m!=PointsPlusNormals && m!=Splatting && m!=ShadedPoints); }
     // OpenGL drawing in a display list
-    virtual void draw() const {}
-    virtual void draw(CGAL::Three::Viewer_interface*) const;
-    virtual void drawPoints(CGAL::Three::Viewer_interface*) const;
-    virtual void drawEdges(CGAL::Three::Viewer_interface* viewer) const;
-    void invalidateOpenGLBuffers();
-    bool isFinite() const { return true; }
-    bool isEmpty() const;
-    void compute_bbox() const;
+    virtual void draw() const Q_DECL_OVERRIDE{}
+    virtual void draw(CGAL::Three::Viewer_interface*) const Q_DECL_OVERRIDE;
+    virtual void drawPoints(CGAL::Three::Viewer_interface*) const Q_DECL_OVERRIDE;
+    virtual void drawEdges(CGAL::Three::Viewer_interface* viewer) const Q_DECL_OVERRIDE;
+    void invalidateOpenGLBuffers() Q_DECL_OVERRIDE;
+    bool isFinite() const Q_DECL_OVERRIDE{ return true; }
+    bool isEmpty() const Q_DECL_OVERRIDE;
+    void compute_bbox() const Q_DECL_OVERRIDE;
 
     void new_vertex(const double&, const double&, const double&);
     void new_triangle(const std::size_t, const std::size_t, const std::size_t);

--- a/Polyhedron/demo/Polyhedron/Scene_polyhedron_item.h
+++ b/Polyhedron/demo/Polyhedron/Scene_polyhedron_item.h
@@ -81,7 +81,7 @@ public:
     // Indicate if rendering mode is supported
     virtual bool supportsRenderingMode(RenderingMode m) const Q_DECL_OVERRIDE;
     // Points/Wireframe/Flat/Gouraud OpenGL drawing in a display list
-    void draw() const {}
+    void draw() const Q_DECL_OVERRIDE{}
     virtual void draw(CGAL::Three::Viewer_interface*) const Q_DECL_OVERRIDE;
     virtual void drawEdges() const Q_DECL_OVERRIDE{}
     virtual void drawEdges(CGAL::Three::Viewer_interface* viewer) const Q_DECL_OVERRIDE;

--- a/Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.h
+++ b/Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.h
@@ -38,24 +38,24 @@ public:
 
   ~Scene_surface_mesh_item();
 
-  Scene_surface_mesh_item* clone() const;
-  void draw(CGAL::Three::Viewer_interface *) const;
-  void drawEdges(CGAL::Three::Viewer_interface *) const;
-  void drawPoints(CGAL::Three::Viewer_interface *) const;
+  Scene_surface_mesh_item* clone() const Q_DECL_OVERRIDE;
+  void draw(CGAL::Three::Viewer_interface *) const Q_DECL_OVERRIDE;
+  void drawEdges(CGAL::Three::Viewer_interface *) const Q_DECL_OVERRIDE;
+  void drawPoints(CGAL::Three::Viewer_interface *) const Q_DECL_OVERRIDE;
 
-  bool supportsRenderingMode(RenderingMode m) const;
-  bool isFinite() const { return true; }
-  bool isEmpty() const;
-  Bbox bbox() const;
-  QString toolTip() const;
+  bool supportsRenderingMode(RenderingMode m) const Q_DECL_OVERRIDE;
+  bool isFinite() const Q_DECL_OVERRIDE { return true; }
+  bool isEmpty() const Q_DECL_OVERRIDE;
+  Bbox bbox() const Q_DECL_OVERRIDE;
+  QString toolTip() const Q_DECL_OVERRIDE;
 
   SMesh* polyhedron();
   const SMesh* polyhedron() const;
-  void compute_bbox()const;
+  void compute_bbox()const Q_DECL_OVERRIDE;
   void standard_constructor(SMesh *sm);
 public Q_SLOTS:
   void itemAboutToBeDestroyed(Scene_item *) Q_DECL_OVERRIDE;
-  virtual void selection_changed(bool);
+  virtual void selection_changed(bool) Q_DECL_OVERRIDE;
 protected:
   friend struct Scene_surface_mesh_item_priv;
   Scene_surface_mesh_item_priv* d;

